pallets/subtensor/src/swap.rs

```diff
@@ -0,0 +1,445 @@
+use super::*;
+use frame_support::{storage::IterableStorageDoubleMap, weights::Weight};
+use sp_core::Get;
+
+impl<T: Config> Pallet<T> {
+    /// Swaps the hotkey of a coldkey account.
+    ///
+    /// # Arguments
+    ///
+    /// * `origin` - The origin of the transaction, and also the coldkey account.
+    /// * `old_hotkey` - The old hotkey to be swapped.
+    /// * `new_hotkey` - The new hotkey to replace the old one.
+    ///
+    /// # Returns
+    ///
+    /// * `DispatchResultWithPostInfo` - The result of the dispatch.
+    ///
+    /// # Errors
+    ///
+    /// * `NonAssociatedColdKey` - If the coldkey does not own the old hotkey.
+    /// * `HotKeySetTxRateLimitExceeded` - If the transaction rate limit is exceeded.
+    /// * `NewHotKeyIsSameWithOld` - If the new hotkey is the same as the old hotkey.
+    /// * `HotKeyAlreadyRegisteredInSubNet` - If the new hotkey is already registered in the subnet.
+    /// * `NotEnoughBalanceToPaySwapHotKey` - If there is not enough balance to pay for the swap.
+    pub fn do_swap_hotkey(
+        origin: T::RuntimeOrigin,
+        old_hotkey: &T::AccountId,
+        new_hotkey: &T::AccountId,
+    ) -> DispatchResultWithPostInfo {
+        let coldkey = ensure_signed(origin)?;
+
+        let mut weight = T::DbWeight::get().reads(2);
+
+        ensure!(old_hotkey != new_hotkey, Error::<T>::NewHotKeyIsSameWithOld);
+        ensure!(
+            !Self::is_hotkey_registered_on_any_network(new_hotkey),
+            Error::<T>::HotKeyAlreadyRegisteredInSubNet
+        );
+
+        weight.saturating_accrue(T::DbWeight::get().reads_writes(2, 0));
+        ensure!(
+            Self::coldkey_owns_hotkey(&coldkey, old_hotkey),
+            Error::<T>::NonAssociatedColdKey
+        );
+
+        let block: u64 = Self::get_current_block_as_u64();
+        ensure!(
+            !Self::exceeds_tx_rate_limit(Self::get_last_tx_block(&coldkey), block),
+            Error::<T>::HotKeySetTxRateLimitExceeded
+        );
+
+        weight
+            .saturating_accrue(T::DbWeight::get().reads((TotalNetworks::<T>::get() + 1u16) as u64));
+
+        let swap_cost = Self::get_hotkey_swap_cost();
+        log::info!("Swap cost: {:?}", swap_cost);
+
+        ensure!(
+            Self::can_remove_balance_from_coldkey_account(&coldkey, swap_cost),
+            Error::<T>::NotEnoughBalanceToPaySwapHotKey
+        );
+        let actual_burn_amount = Self::remove_balance_from_coldkey_account(&coldkey, swap_cost)?;
+        Self::burn_tokens(actual_burn_amount);
+
+        Self::swap_owner(old_hotkey, new_hotkey, &coldkey, &mut weight)?;
+        Self::swap_total_hotkey_stake(old_hotkey, new_hotkey, &mut weight)?;
+        Self::swap_delegates(old_hotkey, new_hotkey, &mut weight)?;
+        Self::swap_last_tx_block(old_hotkey, new_hotkey, &mut weight)?;
```

Comment:
Why do we need to swap LstTxBlock if we're setting a new value of it at the end of this extrinsic anyway?

Reply:
Good catch will take it out

Reply:
done